src/compas/geometry/_intersections/mesh_intersections.py

```
@@ -0,0 +1,96 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from compas.geometry import Point
+from compas.geometry import length_vector
+from compas.geometry import subtract_vectors
+from compas.geometry import intersection_line_triangle
+from compas.geometry import intersection_segment_plane
+
+__all__ = [
+    'intersection_mesh_line',
+    'intersection_mesh_plane',
+    'mesh_vertices_to_points',
+]
+
+
+def intersection_mesh_line(mesh, line):
+    """Compute intersection between mesh faces and line. After one single intersection, stops searching for more.
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+    line : compas.geometry.Line
+
+    Returns
+    -------
+    Point : compas.geometry.Point
+    """
+    for fkey in list(mesh.faces()):
+        vertex_keys = mesh.face_vertices(fkey)
+        if not vertex_keys:
+            continue
+        vertices = [mesh.vertex_attributes(vkey, 'xyz') for vkey in vertex_keys]
+        if len(vertex_keys) not in (3, 4):
+            continue
```

Could this be simplified to 
```
if not (vertex_keys and len(vertex_keys) in (3,4)):
    continue
vertices = [mesh.vertex_atrributes(vkey, 'xyz') for vkey in vertex_keys]
```

---

well or a condition that fails early (the `and` will require both conditions to be checked at all times)
```python
if not vertex_keys or len(vertex_keys) not in (3, 4):
    continue
```